Duplicati/UnitTest/BasicSetupHelper.cs

```diff
@@ -153,6 +158,60 @@ public virtual void TearDown()
             }
         }
 
+        /// <summary>
+        /// Alternative to System.IO.Compression.ZipFile.ExtractToDirectory()
+        /// that handles long paths.
+        /// </summary>
+        protected static void ZipFileExtractToDirectory(string sourceArchiveFileName, string destinationDirectoryName)
+        {
+            if (Platform.IsClientWindows)
+            {
+                // Handle long paths under Windows by extracting to a
+                // temporary file and moving the resulting file to the
+                // actual destination using functions that support
+                // long paths.
+                using (var archive = ZipFile.OpenRead(sourceArchiveFileName))
+                {
+                    foreach (var entry in archive.Entries)
+                    {
+                        // By the ZIP spec, directories end in a forward slash
+                        var isDirectory = entry.FullName.EndsWith("/");
+                        // Convert forward slashes to the Windows separator
+                        var fullNamePath = entry.FullName.Replace("/", Util.DirectorySeparatorString);
+                        var destination = systemIO.PathCombine(destinationDirectoryName, fullNamePath);

```

Comment:
Should we use `PathGetFullPath` to convert the destination path to an absolute path?  The current working directory is unclear if `destinationDirectoryName` is a relative path.

If that works, then it might be possible to remove the conversion of `/` to `\`, since I think `PathGetFullPath` should convert all the separators to the standard one.

Reply:
I agree it needs a call to `PathGetFullPath()` and I can do that.

As far as the slashes, the problem I encountered is that `Path.GetFullPath()` will strip trailing dots and spaces unless it's prefixed with "`\\?\`".  But once you add the prefix, that disables the slash fix-up.  `SystemIOWindows.PathGetFullPath()` adds the "`\\?\`" to handle the trailing dots and spaces, so the forward slashes remain and the resulting path then trips up `DirectoryCreate()`.

One alternative would be to add the slash fix-up code right into `SystemIOWindows.PathGetFullPath()`, or even into `PrefixWithUNC()` itself.  Then the last two lines in the snippet above can turn into what you were proposing:
```
var destination =
    systemIO.PathGetFullPath(systemIO.PathCombine(destinationDirectoryName, entry.FullName));
```

Reply:
> One alternative would be to add the slash fix-up code right into SystemIOWindows.PathGetFullPath()

That would be nice so that `SystemIOWindows.PathGetFullPath` is closer in behavior to the built-in `Path.GetFullPath`.

Reply:
I've pushed a change that adds the slash fix-up code to `SystemIOWindows.PathGetFullPath()`.